homeassistant/components/lock/__init__.py

```diff
@@ -90,7 +90,7 @@ async def _async_lock(entity: LockEntity, service_call: ServiceCall) -> None:
     code: str = service_call.data.get(ATTR_CODE, "")
     if entity.code_format_cmp and not entity.code_format_cmp.match(code):
         raise ValueError(
-            f"Code '{code}' for locking {entity.name} doesn't match pattern {entity.code_format}"
+            f"Code '{code}' for locking {entity.entity_id} doesn't match pattern {entity.code_format}"
```

Comment:
This PR doesn't really motivate this change, so I'm not sure why this change is needed?

This will show up in the UI, the entity ID is not considered a UI-friendly part. In daily use, one will see the name, so why not show the name?

Reply:
I probably took a bit of shortcut on the PR description so allow me to update it but in essence with the addition of `has_entity_name` an entity's friendly name is now put together from device name + entity name. 
This error only shows the later which in the case of Yale for example has not entity.name and therefore shows `None` to the user as the name. 

My other proposal was to make `friendly_name` a property but as the user can always change names etc. I think the only reliable source to present correctly to the user is either `entity_id` or lookup the correct name each time which might be excessive for this. 

What do you think?

Reply:
Updated the PR description properly to show the issue on hand.
This might be an issue in more places too perhaps for other platforms (which doesn't necessarily know how the name is constructed)

Reply:
I'll mark this PR as draft for a moment, I've reached out to other core maintainers to see if we have a consensus on this (as I'm seeing more of these cases).

../Frenck

Reply:
Ok. Thinking about it maybe it would be easier to have a small helper to get the proper name and use if for user-messages.

Reply:
Discussed this today. This is good for now, while this shows up to the frontend, we also have other ideas for that (which include translations and new error types).

Thanks for the PR 👍 

../Frenck